integration/container/remove_test.go

```diff
@@ -0,0 +1,101 @@
+package container // import "github.com/docker/docker/integration/container"
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/integration/internal/request"
+	"github.com/docker/docker/internal/testutil"
+	"github.com/gotestyourself/gotestyourself/fs"
+	"github.com/gotestyourself/gotestyourself/poll"
+	"github.com/gotestyourself/gotestyourself/skip"
+	"github.com/stretchr/testify/require"
+)
+
+func getPrefixAndSlashFromDaemonPlatform() (prefix, slash string) {
+	if testEnv.OSType == "windows" {
+		return "c:", `\`
+	}
+	return "", "/"
+}
+
+func TestRemoveContainerWithRemovedVolume(t *testing.T) {
+	skip.If(t, !testEnv.IsLocalDaemon())
+
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := request.NewAPIClient(t)
+
+	prefix, slash := getPrefixAndSlashFromDaemonPlatform()
+
+	tempDir := fs.NewDir(t, "test-rm-container-with-removed-volume-", fs.WithMode(0755))
+	defer tempDir.Remove()
+
+	cID := container.Run(t, ctx, client, container.WithCmd("true"), func(c *container.TestContainerConfig) {
+		c.HostConfig.Binds = []string{tempDir.Path() + ":" + prefix + slash + "test"}
+	})
+	poll.WaitOn(t, containerIsInState(ctx, client, cID, "exited"), poll.WithDelay(100*time.Millisecond))
+
+	err := os.RemoveAll(tempDir.Path())
+	require.NoError(t, err)
+
+	err = client.ContainerRemove(ctx, cID, types.ContainerRemoveOptions{
+		RemoveVolumes: true,
+	})
+	require.NoError(t, err)
```

Comment:
This is just testing that removing a container doesn't error when the bind mount host path doesn't exist anymore?

I'm not sure about this test, seems like it should be a unit test, it's not testing a behaviour of the system.

Reply:
Thanks @dnephin I found the original issue associated with this test and add a reference to "#5244: `docker rm` fails if bind dir doesn't exist anymore". The test case has been extended to validate that at the end the container has actually been removed (with inspect).